File: modulo_web/tests/Unit/RfidGeneratorTest.php
<?php

namespace Tests\Unit;

use App\Support\RfidGenerator;
use PHPUnit\Framework\Attributes\Test;
use Tests\TestCase;

class RfidGeneratorTest extends TestCase
{
    #[Test]
    public function it_generates_valid_cattle_tags()
    {
        $tag = RfidGenerator::generateCattleTag();
        $this->assertTrue(RfidGenerator::isValid($tag));
        $this->assertStringStartsWith('C', $tag);
    }

    #[Test]
    public function it_generates_valid_vet_tags()
    {
        $tag = RfidGenerator::generateVetTag();
        $this->assertTrue(RfidGenerator::isValid($tag));
        $this->assertStringStartsWith('V', $tag);
    }

    #[Test]
    public function it_validates_various_tags()
    {
        // Valid tags
        $this->assertTrue(RfidGenerator::isValid('C12345'));
        $this->assertTrue(RfidGenerator::isValid('VABC12345678901')); // 16 chars

        // Specialized checks
        $this->assertTrue(RfidGenerator::isCattleTag('C12345'));
        $this->assertFalse(RfidGenerator::isCattleTag('V12345'));
        $this->assertTrue(RfidGenerator::isVetTag('V12345'));
        $this->assertFalse(RfidGenerator::isVetTag('C12345'));

        // Invalid tags
        $this->assertFalse(RfidGenerator::isValid(''));
        $this->assertFalse(RfidGenerator::isValid(null));
        $this->assertFalse(RfidGenerator::isValid('A123'));
        $this->assertFalse(RfidGenerator::isValid('C')); // Too short
        $this->assertFalse(RfidGenerator::isValid('C12345678901234567')); // Too long (17 chars)
        $this->assertFalse(RfidGenerator::isValid('C123-456'));
    }
}

File: modulo_desktop/src/test/java/com/cattlerfid/util/RfidGeneratorTest.java
package com.cattlerfid.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class RfidGeneratorTest {

    @Test
    void should_generate_valid_cattle_tag() {
        String tag = RfidGenerator.generateCattleTag();
        assertTrue(RfidGenerator.isValid(tag), "Generated cattle tag should be valid");
        assertTrue(tag.startsWith("C"), "Cattle tag should start with C");
    }

    @Test
    void should_generate_valid_vet_tag() {
        String tag = RfidGenerator.generateVetTag();
        assertTrue(RfidGenerator.isValid(tag), "Generated vet tag should be valid");
        assertTrue(tag.startsWith("V"), "Vet tag should start with V");
    }

    @Test
    void should_validate_various_tags() {

        assertTrue(RfidGenerator.isValid("C12345"), "Short valid tag");
        assertTrue(RfidGenerator.isValid("VABC12345678901"), "Long valid tag (16 chars)");

        assertTrue(RfidGenerator.isCattleTag("C12345"), "Should be cattle tag");
        assertFalse(RfidGenerator.isCattleTag("V12345"), "Should not be cattle tag");
        assertTrue(RfidGenerator.isVetTag("V12345"), "Should be vet tag");
        assertFalse(RfidGenerator.isVetTag("C12345"), "Should not be vet tag");

        assertFalse(RfidGenerator.isValid(""), "Empty tag");
        assertFalse(RfidGenerator.isValid(null), "Null tag");
        assertFalse(RfidGenerator.isValid("A123"), "Wrong prefix");
        assertFalse(RfidGenerator.isValid("C"), "Too short");
        assertFalse(RfidGenerator.isValid("C1234567890123456"), "Too long (17 chars)");
        assertFalse(RfidGenerator.isValid("C123-456"), "Invalid characters");
    }
}
